README.md

```diff
@@ -63,14 +63,12 @@ installed in order to build the frontend assets.
 You can directly use the `go` tool to download and install the `prometheus`
 and `promtool` binaries into your `GOPATH`:
 
-    $ GO111MODULE=on go get github.com/prometheus/prometheus/cmd/...
+    $ GO111MODULE=on go install github.com/prometheus/prometheus/cmd/...@latest
```

Comment:
Curious, does this actually work for you? I get:

```
GO111MODULE=on go install github.com/prometheus/prometheus/cmd/...@latest
go: downloading github.com/prometheus/prometheus v1.8.2
go: downloading github.com/prometheus/prometheus v2.5.0+incompatible
go: finding module for package github.com/mwitkow/go-conntrack
[...]
go: finding module for package github.com/Azure/azure-sdk-for-go/arm/compute
../../../../pkg/mod/github.com/prometheus/prometheus@v2.5.0+incompatible/discovery/azure/azure.go:24:2: module github.com/Azure/azure-sdk-for-go@latest found (v56.0.0+incompatible), but does not contain package github.com/Azure/azure-sdk-for-go/arm/compute
../../../../pkg/mod/github.com/prometheus/prometheus@v2.5.0+incompatible/discovery/azure/azure.go:25:2: module github.com/Azure/azure-sdk-for-go@latest found (v56.0.0+incompatible), but does not contain package github.com/Azure/azure-sdk-for-go/arm/network
```

With Go 1.16.4 linux/amd64.

Reply:
It works for me without the `@latest`.

Reply:
`...` has to be replaced with either `prometheus` or `promtool`. Works for me.

Reply:
Oh... that is not clear though, as `...` is also valid package selector syntax in Go.

Reply:
Is it? Interesting. `@latest` was requested by @SuperQ here: https://github.com/prometheus/node_exporter/pull/2084

Reply:
Yep, see https://pkg.go.dev/cmd/go#hdr-Package_lists_and_patterns:

> An import path is a pattern if it includes one or more "..." wildcards, each of which can match any string, including the empty string and strings containing slashes. Such a pattern expands to all package directories found in the GOPATH trees with names matching the patterns.

So it would select all packages in subdirectories of `github.com/prometheus/prometheus/cmd/` in this case, which wouldn't actually be bad because it would (IIUC) just install both `promtool` and `prometheus`.

However, even if I do substitute `prometheus` in there, it still fails to build when using `@latest` (getting the same error as before).

Reply:
```
❯ GO111MODULE=on go install github.com/prometheus/prometheus/cmd/...
go: warning: "github.com/prometheus/prometheus/cmd/..." matched no packages
```

When I use `...@latest` I'm getting the same error as you:
```
go/pkg/mod/github.com/prometheus/prometheus@v2.5.0+incompatible/discovery/azure/azure.go:24:2: module github.com/Azure/azure-sdk-for-go@latest found (v56.0.0+incompatible), but does not contain package github.com/Azure/azure-sdk-for-go/arm/compute
go/pkg/mod/github.com/prometheus/prometheus@v2.5.0+incompatible/discovery/azure/azure.go:25:2: module github.com/Azure/azure-sdk-for-go@latest found (v56.0.0+incompatible), but does not contain package github.com/Azure/azure-sdk-for-go/arm/network
```

Reply:
> ```
>❯ GO111MODULE=on go install github.com/prometheus/prometheus/cmd/...
>go: warning: "github.com/prometheus/prometheus/cmd/..." matched no packages
>```

Heh, it only works when you're in the repo directory for some reason, although it's an absolute package name. Well, who even understands the Go tool anymore since Go introduced modules... :laughing: :sob: 

And yeah, when adding a concrete binary name, it says it requires a version like `@latest`, but then it breaks when adding that with the error we now both got. Not sure exactly why, so if anyone has time to look into that, it would be great (but don't feel obligated to).